src/commands/publish/mod.rs

```diff
@@ -37,6 +37,11 @@ pub fn publish(
     validate_worker_name(&target.name)?;
 
     if let Some(site_config) = target.site.clone() {
+        if let Some(route) = &target.route {
+            if !route.ends_with("*") {
+                failure::bail!("The route in your wrangler.toml should have a trailing * to apply the Worker on every path.\nroute = {}*", route)
```

Comment:
I'd use `message::warn` and have this message clarify that without the trailing `*`, your site may not work as expected.

Reply:
i agree, i don't think bailing is necessarily the right move.

Reply:
Yeah - if you see my original comment on the PR, I wasn't sure if we should bail or warn here. The reason I opted for bailing is because I'm 99% sure that it just breaks your site no matter what, and if you do this the first time you're migrating your site from something else, your site will go down, even if you see the warning. I'm willing to change it to a warning, but I'm really not sure we want to go that "route" 😛

Reply:
#iseewhatyoudidthere

my kneejerk reaction is always to say "listen, we don't necessarily know what a user plans on doing with this". However in this case, you make a pretty strong point that a worker that does not use a wild card route here will likely never serve anything other than the root `index.html`, which would be kind of useless.

either way though, i think gabbi's warning of "might not work as expected" is important.

Reply:
even if they were only serving `index.html` and nothing else (which is highly highly unlikely), it would still be in their best interest to have `/*` - if they have other workers that are more specific elsewhere, they will take precedence. I really really don't think we're going to have users come complaining that they can't deploy a worker without the asterisk aside from the usual complaint that its confusing why its necessary at all, which we aren't solving for currently.

Reply:
you've changed my mind. i do like the wording "might not work as expected".

Reply:
@ashleygwilliams I'd love to get your input on this